vowpalwabbit/stagewise_poly.cc

```diff
@@ -269,8 +269,8 @@ void sort_data_update_support(stagewise_poly &poly)
   for (uint64_t i = 0; i != poly.all->length(); ++i)
   { uint64_t wid = stride_shift(poly, i);
     if (!parent_get(poly, wid) && wid != constant_feat_masked(poly))
-    { float wval = (fabsf(poly.all->reg.weight_vector[wid])
-                    * poly.all->reg.weight_vector[poly.all->normalized_idx + (wid)])
+    { float weightsal = (fabsf(poly.all->weights->operator[](wid))
```

Comment:
I'm writing some notes as I work through the code for reference (don't act on them, I'm acting).  
The use of operator[] is bleck.  The basic problem is that we have weight_parameters* weights instead of weight_parameters weights.  I don't see a good reason for a pointer dereference, so I'm going to remove it.